### PR TITLE
feat: add helper text API

### PR DESCRIPTION
**Pull request by DiegoCardoso**
- Add helper text to `vaadin-radio-group` as string attribute and name slotted
- Add demo for the new property and new theme variant
- Add unit and visual tests

 Part of #149

### CODE REVIEW

**Comment by DiegoCardoso** [comment]
> Please rebase this on top of 1.4 branch and switch the base branch for the PR accordingly.

Done.

File: src/vaadin-radio-group.html [hunk 1]
```diff
@@ -470,13 +502,43 @@
...
         /** @private */
         _labelChanged(label) {
-          if (label) {
-            this.setAttribute('has-label', '');
+          this._setOrToggleAttribute('has-label', label === 0 || !!label);
+        }
+
+        /** @private */
+        _errorMessageChanged(errorMessage) {
+          this._setOrToggleAttribute('has-error-message', errorMessage === 0 || !!errorMessage);
```
Comment (web-padawan) [inline src/vaadin-radio-group.html:510]: Why do we check for 0 here, is it a special value?

File: src/vaadin-radio-group.html [hunk 2]
```diff
@@ -470,13 +502,43 @@
 
         /** @private */
         _labelChanged(label) {
-          if (label) {
-            this.setAttribute('has-label', '');
+          this._setOrToggleAttribute('has-label', label === 0 || !!label);
```
Comment (web-padawan) [inline src/vaadin-radio-group.html:505]: Why do we check for 0 here, is it a special value?

File: src/vaadin-radio-group.html [hunk 3]
```diff
@@ -470,13 +502,43 @@
...
+        }
+
+        /** @private */
+        _errorMessageChanged(errorMessage) {
+          this._setOrToggleAttribute('has-error-message', errorMessage === 0 || !!errorMessage);
+        }
+
+        /** @private */
+        _helperTextChanged(helperText) {
+          this._setOrToggleAttribute('has-helper', helperText === 0 || !!helperText);
```
Comment (web-padawan) [inline src/vaadin-radio-group.html:515]: Why do we check for 0 here, is it a special value?

File: src/vaadin-radio-group.html [hunk 4]
```diff
@@ -470,13 +502,43 @@
...
 
+        /** @private */
+        _setOrToggleHasHelperAttribute() {
+          const slottedNodes = this.shadowRoot.querySelector(`[name="helper"]`).assignedNodes();
+          // Only has slotted helper if not a text node
+          // Text nodes are added by the helperText prop and not the helper slot
+          // The filter is added due to shady DOM triggering this slotchange event on helperText prop change
+          this._hasSlottedHelper = slottedNodes.filter(node => node.nodeType !== 3).length > 0;
+
+          this._setOrToggleAttribute('has-helper', this._hasSlottedHelper ? 'slotted' : this.helperText === 0 || !!this.helperText);
```
Comment (web-padawan) [inline src/vaadin-radio-group.html:539]: Why do we check for 0 here, is it a special value?

File: src/vaadin-radio-group.html (hunk 3 above)
Comment (DiegoCardoso) [reply src/vaadin-radio-group.html:515]: Done.

File: src/vaadin-radio-group.html (hunk 1 above)
Comment (DiegoCardoso) [reply src/vaadin-radio-group.html:510]: Done.

File: src/vaadin-radio-group.html (hunk 2 above)
Comment (DiegoCardoso) [reply src/vaadin-radio-group.html:505]: Done.

File: src/vaadin-radio-group.html (hunk 4 above)
Comment (DiegoCardoso) [reply src/vaadin-radio-group.html:539]: Done.

**Pull request merged**